dali/test/python/test_utils.py

```diff
@@ -435,3 +435,98 @@ def np_type_to_dali(type):
         np.float64: types.FLOAT64,
     }
     return np_types_to_dali_dict[type]
+
+
+class AverageMeter(object):
+    """Computes and stores the average and current value"""
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self.val = 0
+        self.avg = 0
+        self.sum = 0
+        self.count = 0
+        self.avg_last_n = 0
+        self.max_val = 0
+
+    def update(self, val, n=1):
+        self.val = val
+        self.max_val = max(self.max_val, val)
+        self.sum += val * n
+        self.count += n
+        self.avg = self.sum / self.count
+
+
+class FilesDataSet(object):
+
+    def __init__(self, data_path):
+        files = self.get_files_list(data_path)
+        self.data_path = data_path
+        self.classes_no = len(files)
+        self.files_map = {}
+
+        counter = 0
+        for class_no, (_, samples) in enumerate(files):
+            for sample in samples:
+                self.files_map[counter] = (sample, class_no)
+                counter += 1
+        self.size = counter
+
+    @classmethod
+    def get_files_list(cls, data_path):
+        dirs = [
+            (dir_name, [
+                file_path for file_path
+                in map(lambda name: os.path.join(dir_path, name), os.listdir(dir_path))
+                if os.path.isfile(file_path)
+            ])
+            for (dir_name, dir_path)
+            in map(lambda name: (name, os.path.join(data_path, name)), os.listdir(data_path))
+            if os.path.isdir(dir_path)
+        ]
+        dirs.sort(key=lambda dir_files: dir_files[0])
+        for _, files in dirs:
+            files.sort()
+        return dirs
+
+    def __len__(self):
+        return self.size
+
+    def __getitem__(self, i):
+        if i >= self.size:
+            raise StopIteration
+        return self.files_map[i]
+
+
+class SampleLoader(object):
+
+    DATA_SET = FilesDataSet
+
+    def __init__(self, data_path, eager=False):
+        self.data_path = data_path
+        self._data_set = None
+        # If there's a state in the dataset that should be shared between workers
+        # you might want to initialize it eagerly
+        if eager:
+            self._data_set = self.create_data_set()
+
+    def __getattr__(self, key):
+        if key == 'data_set':
+            import_numpy()
+            self._data_set = self._data_set or self.create_data_set()
+            self.__dict__['data_set'] = self._data_set
+            return self._data_set
+        raise AttributeError
+
+    def create_data_set(self):
+        return self.DATA_SET(data_path=self.data_path)
+
+    def read_file(self, file_path):
+        with open(file_path, 'rb') as f:
+            return f.read()
+
+    def __call__(self, sample_info):
+        file_path, class_no = self.data_set[sample_info.idx_in_epoch]
```

Comment:
I don't know if we should accept sample_info or just idx_in_epoch.

Reply:
In per sample mode but without parallelization callback already accepts sample_info so I wanted to be in line with that behavior.

Reply:
OK